cmd/snap-bootstrap/cmd_initramfs_mounts.go

```diff
@@ -253,6 +255,25 @@ func generateMountsModeRecover(mst *initramfsMountsState, recoverySystem string)
 	return nil
 }
 
+// TODO:UC20: move all of this to a helper in boot?
+// selectPartitionToMount will select the partition to mount at dir, preferring
+// to use efi variables to determine which partition matches the disk we booted
+// the kernel from. If it can't figure out which disk the kernel came from, then
+// it will fallback to mounting via the specified label
+func selectPartitionMatchingKernelDisk(dir, fallbacklabel string) error {
+	// TODO:UC20: should this only run on grade > dangerous? where do we
+	//            get the model at this point?
+	if partuuid, err := bootFindPartitionUUIDForBootedKernelDisk(); err == nil {
```

Comment:
Should we reverse the test to handle the error case and keep non-error in the main flow?

Reply:
hmm, you're right in that the code logic is slightly inverted here from normal Go conventions... I'll switch this around in a followup I think since this branch is amazingly all green